synapse/rest/media/v1/oembed.py

```diff
@@ -12,30 +12,26 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 import logging
+import urllib.parse
 from typing import TYPE_CHECKING, Optional
 
 import attr
 
 from synapse.http.client import SimpleHttpClient
+from synapse.types import JsonDict
+from synapse.util import json_decoder
 
 if TYPE_CHECKING:
     from synapse.server import HomeServer
 
 logger = logging.getLogger(__name__)
 
 
-@attr.s(slots=True, auto_attribs=True)
+@attr.s(slots=True, frozen=True, auto_attribs=True)
 class OEmbedResult:
-    # Either HTML content or URL must be provided.
-    html: Optional[str]
-    url: Optional[str]
-    title: Optional[str]
+    og: JsonDict
```

Comment:
this name is a little bit cryptic; suggest `opengraph` or ideally something that actually says what it is?

`opengraph_metadata`? I don't really have a good handle of what opengraph is, so you may know better.

Reply:
I agree, but this is what is used throughout this code (which is why I used it here).

Reply:
Oh, also the documentation (which is updated by this PR) does include some information on Open Graph, but pretty much it is a JSON representation of metadata about a resource.